src/condition.cpp

```diff
@@ -491,6 +491,35 @@ conditional_t::func f_has_trait( const JsonObject &jo, std::string_view member,
     };
 }
 
+
+conditional_t::func f_can_see_friends( const JsonObject &jo, std::string_view member, bool is_npc )
```

Comment:
This would work better as a math function that returns the number of visible npcs with a kwarg for attitude, similar to `monsters_nearby()`.

Reply:
I specifically wanted it to be members of the same faction because then we can have dialog conditions where an NPC wants you to have a buddy with you for whatever reason. I also want to ensure that this isn't cheesable with static NPCs like the people in the refugee center, who are not your friends and (mostly) would rather you not hang around.

Reply:
I meant that you should move this to a math function instead of having it as a standalone condition. From math, it can be used in more contexts and you can add whatever filtering you like, including faction. `monsters_nearby()` is a good example for similar functionality and moving the code should be fairly straightforward (look [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/20135845605e538bfcc20a3c28b1b4e5aafdb5e1/src/math_parser_diag.cpp#L1272) and [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/20135845605e538bfcc20a3c28b1b4e5aafdb5e1/src/math_parser_diag.cpp#L623-L672) and maybe [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/math_parser_diag.cpp#L22-L52))

Reply:
Done, tested and working. I had to get some help to do that. The code for math stuff is super confusing compared to the condition code.

Reply:
Thanks; I appreciate it. For extra credit, consider adding a test unit too. It'd be pretty similar to the one for `monsters_nearby()` <https://github.com/CleverRaven/Cataclysm-DDA/blob/master/tests/eoc_test.cpp#L602-L638>

What parts were confusing? You can ping me on discord if it's elaborate